File: src/main/java/entities/utilisateur.java
package entities;

public  class utilisateur {
    private static int id;
    private static String nom;
    private static String prenom;
    private static UserRole role;
    private int numeroDeTelephone;
    private static String email;
    private String motDePasse;
    private  String photo_d;
    public utilisateur( int id,String nom, String prenom, String email, String motDePasse, UserRole role, int numeroDeTelephone, String photo_d) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.role = role;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.motDePasse = motDePasse;
        this.photo_d = photo_d;
    }

    public utilisateur(String nom, String prenom, UserRole role, int numeroDeTelephone, String email, String motDePasse, String photo_d) {
        this.nom = nom;
        this.prenom = prenom;
        this.role = role;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.motDePasse = motDePasse;
        this.photo_d = photo_d;
    }

    public utilisateur(int id) {
        this.id = id;
    }

    public utilisateur() {

    }

    public utilisateur(int id, String nom, String prenom, UserRole role, int numeroDeTelephone, String email, String motDePasse) {
        this.id = id;
        this.nom = nom;
        this.prenom = prenom;
        this.role = role;
        this.numeroDeTelephone = numeroDeTelephone;
        this.email = email;
        this.motDePasse = motDePasse;
    }

    public utilisateur(int id, String nom, String prenom, String email, String motDePasse, UserRole role, int numeroDeTelephone) {
  this.id = id;
  this.nom = nom;
  this.prenom = prenom;
  this.email = email;
  this.motDePasse = motDePasse;
  this.role = role;
  this.numeroDeTelephone = numeroDeTelephone;
    }

    @Override
    public String toString() {
        return "utilisateur{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", prenom='" + prenom + '\'' +
                ", role='" + role + '\'' +
                ", numeroDeTelephone=" + numeroDeTelephone +
                ", email='" + email + '\'' +
                ",motDePasse'" + motDePasse + '\'' +
                '}';
    }

    public static int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public static String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public static String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public static UserRole getRole() {
        return role;
    }

    public void setRole(UserRole role) {

        this.role = role;

    }

    private boolean isValidRole(UserRole role) {
        return role.equals("Camp_Owner") || role.equals("Simple_Utilisateur") || role.equals("Admin");
    }
    public void Utilisateur(String nom, UserRole role) {
        this.nom = nom;
        this.role = role;
    }


    public int getNumeroDeTelephone() {
        return numeroDeTelephone;
    }

    public void setNumeroDeTelephone(int numeroDeTelephone) {
        this.numeroDeTelephone = numeroDeTelephone;
    }

    public static String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setMotDePasse(String motDePasse) {
        this.motDePasse = motDePasse;
    }

    public String getMotDePasse() {
        return motDePasse;
    }

    public String getPhoto_d() {
        return photo_d;
    }

    public void setPhoto_d(String photo_d) {
        this.photo_d = photo_d;
    }
}

File: Utilisateur.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Utilisateur
 *
 * @ORM\Table(name="utilisateur")
 * @ORM\Entity
 */
class Utilisateur
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="Nom", type="string", length=100, nullable=false)
     */
    private $nom;

    /**
     * @var string
     *
     * @ORM\Column(name="Prenom", type="string", length=100, nullable=false)
     */
    private $prenom;

    /**
     * @var int
     *
     * @ORM\Column(name="NumeroDeTelephone", type="integer", nullable=false)
     */
    private $numerodetelephone;

    /**
     * @var string
     *
     * @ORM\Column(name="Email", type="string", length=100, nullable=false)
     */
    private $email;

    /**
     * @var string
     *
     * @ORM\Column(name="MotDePasse", type="string", length=11, nullable=false)
     */
    private $motdepasse;

    /**
     * @var string
     *
     * @ORM\Column(name="role", type="string", length=255, nullable=false)
     */
    private $role = '';

    /**
     * @var string
     *
     * @ORM\Column(name="photo_d", type="string", length=255, nullable=false)
     */
    private $photoD;

    /**
     * @var bool|null
     *
     * @ORM\Column(name="actif", type="boolean", nullable=true)
     */
    private $actif = '0';

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): static
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(string $prenom): static
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getNumerodetelephone(): ?int
    {
        return $this->numerodetelephone;
    }

    public function setNumerodetelephone(int $numerodetelephone): static
    {
        $this->numerodetelephone = $numerodetelephone;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): static
    {
        $this->email = $email;

        return $this;
    }

    public function getMotdepasse(): ?string
    {
        return $this->motdepasse;
    }

    public function setMotdepasse(string $motdepasse): static
    {
        $this->motdepasse = $motdepasse;

        return $this;
    }

    public function getRole(): ?string
    {
        return $this->role;
    }

    public function setRole(string $role): static
    {
        $this->role = $role;

        return $this;
    }

    public function getPhotoD(): ?string
    {
        return $this->photoD;
    }

    public function setPhotoD(string $photoD): self
    {
        $this->photoD = $photoD;

        return $this;
    }

    public function isActif(): ?bool
    {
        return $this->actif;
    }

    public function setActif(?bool $actif): static
    {
        $this->actif = $actif;

        return $this;
    }


}
